influxdb3_cache/src/meta_cache/mod.rs

```diff
@@ -851,4 +346,448 @@ mod tests {
+                use_sorted_assert: false,
+            },
+            TestCase {
+                _desc: "or eq predicates on region",
+                sql: "SELECT * FROM meta_cache('cpu') \
+                    WHERE region = 'us-east' OR region = 'ca-east'",
+                expected: &[
+                    "+---------+------+",
+                    "| region  | host |",
+                    "+---------+------+",
+                    "| ca-east | e    |",
+                    "| us-east | a    |",
+                    "| us-east | b    |",
+                    "+---------+------+",
+                ],
+                explain_contains: "MetaCacheExec: predicates=[[0 IN (ca-east,us-east)]] inner=MemoryExec: partitions=1, partition_sizes=[1]",
+                use_sorted_assert: false,
+            },
+            TestCase {
+                _desc: "or eq predicate on host",
+                sql: "SELECT * FROM meta_cache('cpu') \
+                    WHERE host = 'd' OR host = 'e'",
+                expected: &[
+                    "+---------+------+",
+                    "| region  | host |",
+                    "+---------+------+",
+                    "| ca-east | e    |",
+                    "| us-west | d    |",
+                    "+---------+------+",
+                ],
+                explain_contains: "MetaCacheExec: predicates=[[1 IN (d,e)]] inner=MemoryExec: partitions=1, partition_sizes=[1]",
+                use_sorted_assert: false,
+            },
+            TestCase {
+                _desc: "un-grouped host conditions are not handled in predicate pushdown",
+                sql: "SELECT * FROM meta_cache('cpu') \
+                    WHERE region = 'us-east' AND host = 'a' OR host = 'b'",
+                expected: &[
+                    "+---------+------+",
+                    "| region  | host |",
+                    "+---------+------+",
+                    "| us-east | a    |",
+                    "| us-east | b    |",
+                    "+---------+------+",
+                ],
+                explain_contains: "MetaCacheExec: inner=MemoryExec: partitions=1, partition_sizes=[1]",
+                use_sorted_assert: false,
+            },
+            TestCase {
+                _desc: "grouped host conditions are handled in predicate pushdown",
+                sql: "SELECT * FROM meta_cache('cpu') \
+                    WHERE region = 'us-east' AND (host = 'a' OR host = 'b')",
+                expected: &[
+                    "+---------+------+",
+                    "| region  | host |",
+                    "+---------+------+",
+                    "| us-east | a    |",
+                    "| us-east | b    |",
+                    "+---------+------+",
+                ],
+                explain_contains: "MetaCacheExec: predicates=[[0 IN (us-east)], [1 IN (a,b)]] inner=MemoryExec: partitions=1, partition_sizes=[1]",
```

Comment:
These `explain_contains` are really neat :+1:

Reply:
Yes this is super helpful for testing. I originally had the entire `EXPLAIN` output in the test but unfortunately it was flaky since some things differ in CI 😔. There might be some helpers in datafusion that we could use to normalize the `EXPLAIN` output, will need to dig for that.